### PR TITLE
clean up HTTP(S) resources when changing protocols

### PR DESCRIPTION
**Pull request by agau4779**
Addresses #32

### CODE REVIEW

File: pkg/loadbalancers/l7.go [hunk 1]
```diff
@@ -146,13 +152,25 @@ func (l *L7) edgeHop() error {
...
 	if sslConfigured {
 		glog.V(3).Infof("validating https for %v", l.Name)
 		if err := l.edgeHopHttps(); err != nil {
 			return err
 		}
 	}
+
+	// If SSL is disabled, clean up the HTTPS resources that would have been
+	// derived from this L7.
+	if !sslConfigured {
```
Comment (rramkumar1) [inline pkg/loadbalancers/l7.go]: I don't think you need this if condition. If execution get's here, we already know ssl is not configured based on the check in line 160.

File: pkg/loadbalancers/l7.go [hunk 2]
```diff
@@ -146,13 +152,25 @@ func (l *L7) edgeHop() error {
 		if err := l.checkStaticIP(); err != nil {
 			return err
 		}
+	} else {
```
Comment (rramkumar1) [inline pkg/loadbalancers/l7.go]: What's the reason for deleting the IP here?

File: pkg/loadbalancers/l7.go [hunk 3]
```diff
@@ -213,18 +242,10 @@ func (l *L7) Cleanup() error {
...
-
-	tpsName := l.namer.TargetProxy(l.Name, utils.HTTPSProtocol)
-	glog.V(2).Infof("Deleting target https proxy %v", tpsName)
-	if err := utils.IgnoreHTTPNotFound(l.cloud.DeleteTargetHttpsProxy(tpsName)); err != nil {
-		return err
-	}
+	return nil
+}
 
+func (l *L7) deleteSslCerts() error {
```
Comment (rramkumar1) [inline pkg/loadbalancers/l7.go]: can we call this deleteIngressManagedSslCerts()

File: pkg/loadbalancers/l7.go (hunk 2 above)
Comment (agau4779) [reply pkg/loadbalancers/l7.go]: If we don't need an IP, was hoping to delete it to avoid abandoning it. Do we not need it?

Comment (agau4779) [reply pkg/loadbalancers/l7.go]: Removed this line, added todo - in the future we need to consider whether or not to demote the IP to ephemeral

**Pull request closed without merging**